File: app/graphql/mutations/create_message.rb
module Mutations
  class CreateMessage < Mutations::BaseMutation
    graphql_name "CreateMessage"
    # arguments passed to the `resolve` method
    argument :name, String, required: true
    argument :email, String, required: true
    argument :company, String, required: true
    argument :message, String, required: true
    # return type from the mutation
    field :message, Types::MessageType, null: false
    def resolve(args)
      message = MessageMailer.with(args: args).contact_email.deliver_now
      MutationResult.call(
        errors: message.errors
      )
    end
  end
end

File: app/javascript/mutations/createMessage.js
// app/javascript/mutations/createMessage.js

import gql from 'graphql-tag';

const mutation = gql`
  mutation createMessage(
    $name: String!,
    $email: String!,
    $company: String!,
    $message: String!,
  ) {
    createMessage(input: {
      name: $name,
      email: $email,
      company: $company,
      message: $message,
    }) {
      success
      errors
    }
  }
`;

export default function({
  apollo,
  name,
  email,
  company,
  message
}) {
  return apollo.mutate({
    mutation,
    variables: {
      name,
      email,
      company,
      message
    },
  });
}